src/common/transformations/tests/common_optimizations/shared_ops_optimization.cpp

```diff
@@ -53,4 +55,181 @@ TEST_F(TransformationTestsF, SharedSlice) {
+
+        auto concat = std::make_shared<v0::Concat>(OutputVector{concat_0, concat_1}, 0);
+        auto result = std::make_shared<v0::Result>(concat);
+        model = std::make_shared<ov::Model>(ResultVector{result}, ParameterVector{data});
+        manager.register_pass<ov::pass::SharedOpOptimization>();
+    }
+    {
+        auto pre_constant_0 = v0::Constant::create(element::f32, Shape{4}, std::vector<float>{3.14, 42, 0, 14});
+        auto data = std::make_shared<v0::Parameter>(element::f32, PartialShape{-1});
+        auto post_constant = v0::Constant::create(element::f32, Shape{1}, std::vector<float>{3.14});
+
+        auto concat_0 = std::make_shared<v0::Concat>(OutputVector{pre_constant_0, data, post_constant}, 0);
+
+        auto concat = std::make_shared<v0::Concat>(OutputVector{concat_0, concat_0}, 0);
+        auto result = std::make_shared<v0::Result>(concat);
+
+        model_ref = std::make_shared<ov::Model>(ResultVector{result}, ParameterVector{data});
+    }
+}
+
+TEST_F(TransformationTestsF, SharedConcatCheckOpWithResultIsntReplaced) {
+    {
+        auto pre_constant_0 = v0::Constant::create(element::f32, Shape{4}, std::vector<float>{3.14, 42, 0, 14});
+        auto pre_constant_1 = v0::Constant::create(element::f32, Shape{4}, std::vector<float>{3.14, 42, 0, 14});
+        auto data = std::make_shared<v0::Parameter>(element::f32, PartialShape{-1});
+        auto post_constant = v0::Constant::create(element::f32, Shape{1}, std::vector<float>{3.14});
+
+        auto concat_0 = std::make_shared<v0::Concat>(OutputVector{pre_constant_0, data, post_constant}, 0);
+        auto concat_1 = std::make_shared<v0::Concat>(OutputVector{pre_constant_1, data, post_constant}, 0);
+
+        auto result_0 = std::make_shared<v0::Result>(concat_0);
+        auto result_1 = std::make_shared<v0::Result>(concat_1);
+        model = std::make_shared<ov::Model>(ResultVector{result_0, result_1}, ParameterVector{data});
+        manager.register_pass<ov::pass::SharedOpOptimization>();
+    }
+    {
+        auto pre_constant_0 = v0::Constant::create(element::f32, Shape{4}, std::vector<float>{3.14, 42, 0, 14});
+        auto pre_constant_1 = v0::Constant::create(element::f32, Shape{4}, std::vector<float>{3.14, 42, 0, 14});
+        auto data = std::make_shared<v0::Parameter>(element::f32, PartialShape{-1});
+        auto post_constant = v0::Constant::create(element::f32, Shape{1}, std::vector<float>{3.14});
+
+        auto concat_0 = std::make_shared<v0::Concat>(OutputVector{pre_constant_0, data, post_constant}, 0);
+        auto concat_1 = std::make_shared<v0::Concat>(OutputVector{pre_constant_1, data, post_constant}, 0);
+
+        auto result_0 = std::make_shared<v0::Result>(concat_0);
+        auto result_1 = std::make_shared<v0::Result>(concat_1);
+        model_ref = std::make_shared<ov::Model>(ResultVector{result_0, result_1}, ParameterVector{data});
+    }
```

Comment:
This is not needed, [reference model is cloned if not defined](https://github.com/openvinotoolkit/openvino/blob/f470825fafb2a298172de84ef4485184f0c7f167/src/tests/test_utils/common_test_utils/src/ngraph_test_utils.cpp#L61-L63).